coderd/database/queries/activitybump.sql

```diff
@@ -10,12 +10,20 @@ WITH latest AS (
 		workspace_builds.max_deadline::timestamp with time zone AS build_max_deadline,
 		workspace_builds.transition AS build_transition,
 		provisioner_jobs.completed_at::timestamp with time zone AS job_completed_at,
-		(workspaces.ttl / 1000 / 1000 / 1000 || ' seconds')::interval AS ttl_interval
+		(
+			CASE
+				WHEN templates.allow_user_autostop
+				THEN (workspaces.ttl / 1000 / 1000 / 1000 || ' seconds')::interval
+				ELSE (templates.default_ttl / 1000 / 1000 / 1000 || ' seconds')::interval
```

Comment:
Perhaps I'm missing some context, but is the `workspace.ttl` the same as this setting?

<img width="600" alt="image" src="https://github.com/coder/coder/assets/147409/0b2ad2fd-b0c6-4b2e-8d27-018410bd42bd">

If yes, that's IMO quite different from the template one:

<img width="600" alt="image" src="https://github.com/coder/coder/assets/147409/5b1ac5e8-9b16-48d0-9364-1313027dcaeb">

In the screenshots, the workspace autostop does not IMO translate to a TTL and would produce unexpected results given this query logic. But as I said, I may be missing context and perhaps this is hat Deans PR is about (I haven't looked at it).

Reply:
Yeah this makes more sense in the context of the [original PR](https://github.com/coder/coder/pull/10035). I do agree that the current autostop/TTL behaviour has grown unwieldy though.